### DISCUSSION TITLE
BUG: loadable util didn't handle error of observable atom with jotai 2.0

### DISCUSSION THREAD

**Discussion by DornieDortsch** [pmndrs/jotai#1843] [Bug report]
I use `atomWithObservable` with the `loadable` util function. When an error occurs at the obseravale atom, the `loadable` didn't handle it. This works with jotai 1.4 but not with 2.0

Here is a modified Counter example, that show the behaviour:

```ts
const counterSubject = interval(1000).pipe(
  map((i) => {
    if (i === 3) {
      throw Error("test");
    }
    return `Execution #${i}`;
  })
);
const counterAtom = atomWithObservable(() => counterSubject);
const loadableCounterAtom = loadable(counterAtom);

const Counter = () => {
  const [counter] = useAtom(loadableCounterAtom);

  return (
    <>
      {counter.state === "loading" && "loading..."}
      {counter.state === "hasData" && <div>{counter.data as string}</div>}
      {counter.state === "hasError" && <div>{`${counter.error}`}</div>}
    </>
  );
};

export default function App() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>
      <Counter />
    </div>
  );
}
```

https://codesandbox.io/s/react-typescript-forked-forked-qd9gqq?file=/src/App.tsx

**Comment by dai-shi** [comment]
Thanks for the reproduction.
Yeah, it looks like a regression/bug.
Can you or anyone investigate it?
We need a test for it too.

**Reply by barelyhuman** [reply]
@dai-shi 
marking the map function as `async` throws the error and is handled by `loadable` as needed, is that still the needed behaviour?

**Reply by dai-shi** [reply]
Not sure if I follow. Can you elaborate please?

**Reply by barelyhuman** [reply]
As in , `map(async ()=>{})` solves the issue, does that still need fixing?

**Reply by dai-shi** [reply]
I see what you mean. It does fix it.
So, I think I understand the problem. `atomWithObservable` creates an atom that is async at first but sync afterwards. And, `loadable` only cares promise errors. Hm, `loadable` supports non promise values, so it's better to support non promise errors. Let me try.

**Reply by dai-shi** [reply]
#2021 should fix it.
https://codesandbox.io/s/react-typescript-forked-forked-vpdrnw?file=/src/App.tsx

@barelyhuman Thanks so much for the hint.

**Reply by barelyhuman** [reply]
Anytime 😄
